cajita/unit_test/tstLoadBalancer.hpp

```diff
@@ -0,0 +1,96 @@
+/****************************************************************************
+ * Copyright (c) 2018-2021 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cajita_GlobalGrid.hpp>
+#include <Cajita_LoadBalancer.hpp>
+#include <Cajita_Types.hpp>
+#include <Cajita_UniformDimPartitioner.hpp>
+
+#include <gtest/gtest.h>
+
+#include <mpi.h>
+
+#include <array>
+#include <cmath>
+
+using namespace Cajita;
+
+namespace Test
+{
+
+//---------------------------------------------------------------------------//
+double lbQuality( double work )
+{
+    // Quality defined as (wmax-wmin)/(wmax+wmin)
+    // Meaning 0 represents perfectly balanced and 1 empty ranks.
+    double wmax, wmin;
+    MPI_Allreduce( &work, &wmax, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD );
+    MPI_Allreduce( &work, &wmin, 1, MPI_DOUBLE, MPI_MIN, MPI_COMM_WORLD );
+    return ( wmax - wmin ) / ( wmax + wmin );
+}
+
+double lbWork( std::array<double, 6>& verts )
+{
+    // work density function: x+y+z
+    double dx = verts[3] - verts[0];
+    double dy = verts[4] - verts[1];
+    double dz = verts[5] - verts[2];
+    return 0.5 * ( verts[3] * verts[3] - verts[0] * verts[0] ) * dy * dz +
+           0.5 * ( verts[4] * verts[4] - verts[1] * verts[1] ) * dx * dz +
+           0.5 * ( verts[5] * verts[5] - verts[2] * verts[2] ) * dx * dy;
+}
+
+void lbTest3d()
+{
+    std::size_t test_steps = 50; // Number of times to rebalance
+    UniformDimPartitioner partitioner;
+
+    // Global mesh
+    double cell_size = 0.23;
+    std::array<int, 3> global_num_cell = { 47, 38, 53 };
+    std::array<double, 3> global_low_corner = { 1.2, 3.3, -2.8 };
+    std::array<double, 3> global_high_corner = {
+        global_low_corner[0] + cell_size * global_num_cell[0],
+        global_low_corner[1] + cell_size * global_num_cell[1],
+        global_low_corner[2] + cell_size * global_num_cell[2] };
+    auto global_mesh = createUniformGlobalMesh(
+        global_low_corner, global_high_corner, global_num_cell );
+    std::array<bool, 3> is_dim_periodic = { false, false, false };
+
+    // Global grid
+    auto global_grid = createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                         is_dim_periodic, partitioner );
+    int rank;
+    MPI_Comm_rank( MPI_COMM_WORLD, &rank );
+    LoadBalancer<Cajita::UniformMesh<double, 3>> lb( MPI_COMM_WORLD,
```

Comment:
Could add a `createLoadBalancer` to match the others

Reply:
done